### PR TITLE
BUG: Should not raise error in concatenating Series with numpy scalar and tuple names (GH21015)

### LINKED ISSUE #21015: BUG: error in concatting Series with numpy scalar / tuple names
```
In [2]: s1 = pd.Series({'a': 1.5}, name=np.int64(190))

In [3]: s2 = pd.Series([], name=(43, 0))

In [4]: pd.concat([s1, s2])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-5f754290d56c> in <module>()
----> 1 pd.concat([s1, s2])

~/scipy/pandas/pandas/core/reshape/concat.py in concat(objs, axis, join, join_axes, ignore_index, keys, levels, names, verify_integrity, sort, copy)
    224                        verify_integrity=verify_integrity,
    225                        copy=copy, sort=sort)
--> 226     return op.get_result()
    227 
    228 

~/scipy/pandas/pandas/core/reshape/concat.py in get_result(self)
    385             # stack blocks
    386             if self.axis == 0:
--> 387                 name = com._consensus_name_attr(self.objs)
    388 
    389                 mgr = self.objs[0]._data.concat([x._data for x in self.objs],

~/scipy/pandas/pandas/core/common.py in _consensus_name_attr(objs)
     56     name = objs[0].name
     57     for obj in objs[1:]:
---> 58         if obj.name != name:
     59             return None
     60     return name

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

It looks maybe a bit of a strange corner case, but you can get both cases rather easily (the one from selecting from Int64Index axis (not default RangeIndex), the other from MultiIndex axis), for example:

```
In [15]: df1 = pd.DataFrame([[1, 2], [3, 4]], columns=['a', 'b'], index=[0, 1])

In [16]: df2 = pd.DataFrame([[5, 6], [7, 8]], columns=['c', 'd'], index=pd.MultiIndex.from_tuples([(0, 0), (1, 1)]))

In [17]: pd.concat([df1.iloc[0], df2.iloc[0]])
...
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

The reason for the error is that if you compare a tuple and a numpy scalar, it gives an array as result:

```
In [18]: (0, 0) == np.int64(0)
Out[18]: array([ True,  True], dtype=bool)
```

### PR DESCRIPTION
**Pull request by KalyanGokhale**
- [x] closes #21015 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/common.py [hunk 1]
```diff
@@ -55,7 +55,11 @@ def flatten(l):
 def _consensus_name_attr(objs):
     name = objs[0].name
     for obj in objs[1:]:
-        if obj.name != name:
+        name_check = (obj.name != name)
+        if not isinstance(name_check, (bool, np.bool_)):
```
Comment (WillAyd) [inline pandas/core/common.py]: Can you use a try...except block here to catch the ValueError instead of type introspection?

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the PR! Comments below - can you also add a whatsnew note?

File: pandas/core/common.py (hunk 1 above)
Comment (KalyanGokhale) [reply pandas/core/common.py]: @WillAyd thanks - will make the changes. Should I add whatsnew note to v0.23.1?

Comment (WillAyd) [reply pandas/core/common.py]: Yes that works

Comment (KalyanGokhale) [reply pandas/core/common.py]: Done

File: pandas/core/common.py [hunk 2]
```diff
@@ -55,7 +55,10 @@ def flatten(l):
 def _consensus_name_attr(objs):
     name = objs[0].name
     for obj in objs[1:]:
-        if obj.name != name:
+        try:
+            if obj.name != name:
+                return None
```
Comment (jreback) [inline pandas/core/common.py]: do like
```
try:
    if obj.name != name:
        name = None
 except ValueError:
    name = None

return name
```

File: doc/source/whatsnew/v0.23.1.txt [hunk 3]
```diff
@@ -82,7 +82,7 @@ Plotting
 Reshaping
 ^^^^^^^^^
 
--
+- Bug in :func:`concat` where error was raised in concatenating Series with numpy scalar and tuple names (:issue:`21015`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.1.txt]: double backticks around Series, is this *only* a numpy scalar or any scalar?

File: pandas/tests/reshape/test_concat.py [hunk 4]
```diff
@@ -2350,6 +2350,21 @@ def test_concat_datetime_timezone(self):
 
         tm.assert_frame_equal(result, expected)
 
+    def test_concat_series_name_npscalar_tuple(self):
```
Comment (jreback) [inline pandas/tests/reshape/test_concat.py]: this doesn't need to be in the class, it can just be a module level test (e.g. de-dent)

Comment (jreback) [inline pandas/tests/reshape/test_concat.py]: parameterize this on a np.int64 and an int

File: pandas/tests/reshape/test_concat.py [hunk 5]
```diff
@@ -2350,6 +2350,21 @@ def test_concat_datetime_timezone(self):
...
+        # GH21015
+        s1 = pd.Series({'a': 1.5}, name=np.int64(190))
+        s2 = pd.Series([], name=(43, 0))
+        result = pd.concat([s1, s2])
+        expected = pd.Series({'a': 1.5})
+        tm.assert_series_equal(result, expected)
+
+        df1 = pd.DataFrame([[1, 2], [3, 4]], columns=['a', 'b'], index=[0, 1])
```
Comment (jreback) [inline pandas/tests/reshape/test_concat.py]: this is very weird to construct a DataFrame to then take a row from it. Just directly construct the series.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/common.py (hunk 2 above)
Comment (KalyanGokhale) [reply pandas/core/common.py]: @jreback Thanks - will make the changes

File: doc/source/whatsnew/v0.23.1.txt (hunk 3 above)
Comment (KalyanGokhale) [reply doc/source/whatsnew/v0.23.1.txt]: only numpy scalar - as it returns an array on comparison with tuple

File: pandas/tests/reshape/test_concat.py (hunk 5 above)
Comment (KalyanGokhale) [reply pandas/tests/reshape/test_concat.py]: will update - had added the original test case from the issue #21015

File: doc/source/whatsnew/v0.23.1.txt (hunk 3 above)
Comment (jreback) [reply doc/source/whatsnew/v0.23.1.txt]: ok that's fine (but add in an integer in the test as well below)

File: pandas/core/common.py (hunk 2 above)
Comment (KalyanGokhale) [reply pandas/core/common.py]: done

File: pandas/tests/reshape/test_concat.py (hunk 4 above)
Comment (KalyanGokhale) [reply pandas/tests/reshape/test_concat.py]: done - took the test out of class, parameterised as suggested, also actually removed the test case which took a row from DataFrame

File: pandas/tests/reshape/test_concat.py [hunk 6]
```diff
@@ -2487,3 +2487,14 @@ def test_concat_aligned_sort_does_not_raise():
                             columns=[1, 'a'])
     result = pd.concat([df, df], ignore_index=True, sort=True)
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("s1name, s2name", [(np.int64(190), (43, 0)),
```
Comment (WillAyd) [inline pandas/tests/reshape/test_concat.py]: No space between `s1name` and `s2name` here

File: pandas/tests/reshape/test_concat.py [hunk 7]
```diff
@@ -2487,3 +2487,14 @@ def test_concat_aligned_sort_does_not_raise():
...
+                                            (190, (43, 0))])
```
Comment (WillAyd) [inline pandas/tests/reshape/test_concat.py]: Could probably push the tuple on the above line down here - would make the indentation more readable

**Review by WillAyd — CHANGES_REQUESTED** [review]
Very minor style nits but otherwise lgtm

File: pandas/tests/reshape/test_concat.py [hunk 8]
```diff
@@ -2487,3 +2487,14 @@ def test_concat_aligned_sort_does_not_raise():
                             columns=[1, 'a'])
     result = pd.concat([df, df], ignore_index=True, sort=True)
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("s1name,s2name", [
+                        (np.int64(190), (43, 0)), (190, (43, 0))])
```
Comment (WillAyd) [inline pandas/tests/reshape/test_concat.py]: The open bracket is correct but the line below is now over-indented - make sure you set that back accordingly

File: doc/source/whatsnew/v0.23.1.txt [hunk 9]
```diff
@@ -82,7 +82,7 @@ Plotting
 Reshaping
 ^^^^^^^^^
 
--
+- Bug in :func:`concat` where error was raised in concatenating ``Series`` with numpy scalar and tuple names (:issue:`21015`)
```
Comment (WillAyd) [inline doc/source/whatsnew/v0.23.1.txt]: Let's replace \`\`Series\`\` with :class:\`Series\`

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.23.1.txt (hunk 9 above)
Comment (KalyanGokhale) [reply doc/source/whatsnew/v0.23.1.txt]: done

File: pandas/tests/reshape/test_concat.py (hunk 8 above)
Comment (KalyanGokhale) [reply pandas/tests/reshape/test_concat.py]: done - sorry for so many iterations

**Review by WillAyd — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @KalyanGokhale

**Pull request merged**